### PR TITLE
Feature/components/cart

### PR DESCRIPTION
**Pull request by Vomcat**

### CODE REVIEW

File: src/components/Cart/Cart.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  colors,
+} from "styles";
+
+const CartWrapper = styled.div<Pick<CartStyleProps, "variant">>`
+  display: flex;
+  flex-direction: column;
+  gap: ${dimensions.spacing.md}px;
+  ${(props) =>
+    props.variant === "menu" &&
+    `position: absolute; top: 42px; right:0; padding: 50px; width:500px;  background-color: ${colors.white}; border-bottom-left-radius:17px; border-bottom-right-radius:17px;`}
```
Comment (magnowak) [inline src/components/Cart/Cart.tsx]: let's format this into separate lines for each css property so it is more legible

File: src/store/cartSlice.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    totalPrice: number;
+    quantity: number;
+  }[];
+  totalAmount: number;
+  totalQuantity: number;
+}
+
+const initialState: CartSliceState = {
+  items: JSON.parse(localStorage.getItem("cart") as string) || [],
+  totalAmount: JSON.parse(localStorage.getItem("totalAmount") as string) || 0,
```
Comment (magnowak) [inline src/store/cartSlice.ts:17]: "totalPrice"?

File: src/components/layout/tests/ButtonPlus.test.tsx [hunk 3]
```diff
@@ -24,6 +24,6 @@ describe("ButtonPlus component", () => {
 
   test("button minus bg-color", () => {
     const tree = renderer.create(<ButtonPlus variant="grey" />).toJSON();
-    expect(tree).toHaveStyleRule("background-color", "#D2CDCD");
+    expect(tree).toHaveStyleRule("background-color", "#D9D9D9");
```
Comment (magnowak) [inline src/components/layout/tests/ButtonPlus.test.tsx]: refer to styles variable

File: src/interfaces/Cart.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export interface CartStyleProps {
+  variant: "default" | "menu";
+  items: any[];
+}
+
+export interface CartItemProps {
+  id?: number;
```
Comment (magnowak) [inline src/interfaces/Cart.ts]: why is id optional?

File: src/styles/mixins.tsx [hunk 5]
```diff
@@ -21,3 +22,15 @@ export const respondTo = (breakpoints: number) => {
 export const baseShadow = css`
   box-shadow: 0px 4px 4px -2px rgba(0, 0, 0, 0.1);
 `;
+
+export const errorText = css`
+  color: ${colors.red};
+  text-decoration: underline;
+  font-size: 14px;
```
Comment (magnowak) [inline src/styles/mixins.tsx]: refer to style variable

File: src/store/cartSlice.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+interface CartSliceState {
+  items: {
+    id: number;
+    name: string;
+    price: number;
+    totalPrice: number;
```
Comment (magnowak) [inline src/store/cartSlice.ts:8]: shouldn't this be a property of all items and not of single ones?

**Review by magnowak — APPROVED** [review]

File: src/components/Cart/Cart.tsx (hunk 1 above)
Comment (Vomcat) [reply src/components/Cart/Cart.tsx]: Done

File: src/store/cartSlice.ts (hunk 2 above)
Comment (Vomcat) [reply src/store/cartSlice.ts:17]: name "totalPrice" is used in antoher place

File: src/components/layout/tests/ButtonPlus.test.tsx (hunk 3 above)
Comment (Vomcat) [reply src/components/layout/tests/ButtonPlus.test.tsx]: done

File: src/interfaces/Cart.ts (hunk 4 above)
Comment (Vomcat) [reply src/interfaces/Cart.ts]: by mistake, corrected

File: src/styles/mixins.tsx (hunk 5 above)
Comment (Vomcat) [reply src/styles/mixins.tsx]: done

File: src/store/cartSlice.ts (hunk 6 above)
Comment (Vomcat) [reply src/store/cartSlice.ts:8]: we talked about it at the meeting. This is the total price of one item in the cart

**Pull request merged**